website/content/docs/secrets/transit.mdx

```diff
@@ -232,6 +232,57 @@ the proper permission, it can use this secrets engine.
     data, since the process would not be able to get access to the plaintext
     data.
 
+## BYOK
+
+The transit secrets engine supports importing key material from external sources into
+transit keys.
+
+First, the wrapping key needs to be read from transit:
+
+```text
+$ vault read transit/wrapping_key
+```
+
+The wrapping key will be a 4096-bit RSA public key.
+
+Then the wrapping key is used to create the ciphertext input for the `import` endpoint,
+as described below. In the below, the target key refers to the key being imported.
+
+### HSM
+
+If the key is being imported from an HSM that supports PKCS#11, there are
+two possible scenarios:
+
+- If the HSM supports the CKM_RSA_AES_KEY_WRAP mechanism, that can be used to wrap the
+target key using the wrapping key.
+
+- Otherwise, two mechanisms can be combined to wrap the target key. First, an AES key should
+be generated and then used to wrap the target key using the CKM_AES_KEY_WRAP_PAD mechanism.
```

Comment:
Just a note, this mechanism is wrong. Per the [updated PKCS#11 v3.0 spec](http://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/os/pkcs11-curr-v3.0-os.html#_Toc370634387) and the Transit implementation, this should be `CKM_AES_KEY_WRAP_KWP`.